lib/routers/profile/alerts.js

```diff
@@ -0,0 +1,152 @@
+      model: pil,
+      deadline: pil.reviewDate,
+      overdue: pil.reviewOverdue
+    });
+  }
+
+  const raProjects = await raDueQuery(models).where({ licenceHolderId: profile.id });
+
+  raProjects.forEach(project => {
+    alerts.push({
+      type: 'raDue',
+      model: project,
+      deadline: project.raDate,
+      overdue: moment(project.raDate).isBefore(moment())
+    });
+  });
+
+  const ropsProjects = await ropsDueQuery(models, ropsYear).where({ licenceHolderId: profile.id });
+
+  ropsProjects.forEach(project => {
+    alerts.push({
+      type: 'ropDue',
+      model: project,
+      deadline: project.ropsDeadline,
+      overdue: moment(project.ropsDeadline).isBefore(moment()),
+      ropsYear
+    });
+  });
+
+  return alerts;
+};
+
+function isNtcoAtEstablishment(profile, estId) {
+  return !!(profile.roles || []).find(role => role.establishmentId === estId && role.type === 'ntco');
+}
+
+const getEstablishmentAlerts = async (profile, models, ropsYear) => {
+  const alerts = [];
+  const adminAtEstablishments = (profile.establishments || []).filter(e => e.role === 'admin');
+  const ntcoAtEstablishments = (profile.establishments || []).filter(e => isNtcoAtEstablishment(profile, e.id));
+  const pilReviewEstablishments = adminAtEstablishments.concat(ntcoAtEstablishments);
+
+  if (pilReviewEstablishments.length > 0) {
+    const { PIL } = models;
+
+    const pilReviews = await PIL.query()
+      .where({ status: 'active' })
+      .where('reviewDate', '<', moment().add(2, 'months'))
+      .whereIn('establishmentId', pilReviewEstablishments.map(e => e.id));
+
+    pilReviews.forEach(pil => {
+      alerts.push({
+        type: 'pilReview',
+        model: pil,
+        establishmentId: pil.establishmentId,
+        deadline: pil.reviewDate,
+        overdue: moment(pil.reviewDate).isBefore(moment())
+      });
+    });
+  }
+
+  if (adminAtEstablishments.length > 0) {
+    const raProjects = await raDueQuery(models)
+      .whereIn('projects.establishmentId', adminAtEstablishments.map(e => e.id));
+
+    raProjects.forEach(project => {
+      alerts.push({
+        type: 'raDue',
+        model: project,
+        establishmentId: project.establishmentId,
+        deadline: project.raDate,
+        overdue: moment(project.raDate).isBefore(moment())
+      });
+    });
+
+    const ropsProjects = await ropsDueQuery(models, ropsYear)
+      .whereIn('projects.establishmentId', adminAtEstablishments.map(e => e.id));
+
+    ropsProjects.forEach(project => {
+      alerts.push({
+        type: 'ropDue',
+        model: project,
+        establishmentId: project.establishmentId,
+        deadline: project.ropsDeadline,
+        overdue: moment(project.ropsDeadline).isBefore(moment()),
+        ropsYear
+      });
+    });
+  }
+
+  return alerts;
+};
+
+module.exports = () => {
+  const router = Router({ mergeParams: true });
+
+  router.get('/',
+    permissions('profile.alerts', req => ({ profileId: req.profile.id })),
```

Comment:
I've added a new perm `profile.alerts: ['profile:own']` here but it seems a bit superfluous as all external users will be allowed that perm as we're passing their own profile to it.

Having no perm check here sticks a warning in the log for every request... not sure on the best approach.

Reply:
- https://github.com/UKHomeOffice/asl-permissions/pull/138